### DISCUSSION TITLE
Clear button in toolbar takes too long to clear the paper

### DISCUSSION THREAD

**Discussion by kvinita** [clientIO/joint#2591] [Q&A] [answered]
### Introduction

I have used the Kitchen Sink sample code to create a toolbar like below:
![image](https://github.com/clientIO/joint/assets/124018900/8c9eb19a-49d6-4b96-ae2a-12eefcc822fb)

I have initialized toolbar liks:
  ```
initializeToolbar(): void {
    this.toolbarService.create(this.commandManager, this.paperScroller);

    this.toolbarService.toolbar.on({
      'svg:pointerclick': this.openAsSVG.bind(this),
      'png:pointerclick': this.openAsPNG.bind(this),
      'to-front:pointerclick': this.applyOnSelection.bind(this, 'toFront'),
      'to-back:pointerclick': this.applyOnSelection.bind(this, 'toBack'),
      'snapline:change': this.changeSnapLines.bind(this),
      'clear:pointerclick': this.graph.clear.bind(this.graph),
      'print:pointerclick': this.paper.print.bind(this.paper),
      'grid-size:change': this.paper.setGridSize.bind(this.paper),
    });

    this.renderPlugin('.toolbar-container', this.toolbarService.toolbar);
  }
```
When i click on clear button, if the paper has 100+ items then the operation takes too long to complete. Is there any other way to reduce the time taken to clear?

FYI, I have tried using `resetCells([]);` but that did not work.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
> FYI, I have tried using resetCells([]); but that did not work.

What does `did not work` mean?

**Reply by kvinita** [reply]
It created runtime errors

**Reply by kumilingus** [reply]
What errors?

**Reply by kumilingus** [reply]
I need to see the exact error message from the console in order to help you.

`graph.resetCells([])` does not usually throw an error.

You check out our guide on how to write a good technical questions [here](https://github.com/clientIO/joint/wiki/Asking-Technical-Questions).

**Comment by kvinita** [comment, marked as answer]
Apologies for incomplete description. 
The method resetCells() indeed worked.
`'clear:pointerclick': () => this.graph.resetCells([]),`
